tools/src/main/java/org/apache/kafka/tools/consumer/group/ShareGroupCommand.java

```diff
@@ -366,6 +428,70 @@ Entry<Throwable, Map<String, Throwable>> sendDeleteShareGroupOffsetsRequest(Stri
             return new SimpleImmutableEntry<>(topLevelException, topicLevelResult);
         }
 
+        void resetOffsets() throws ExecutionException, InterruptedException {
+            String groupId = opts.options.valueOf(opts.groupOpt);
+            Map<TopicPartition, OffsetAndMetadata> offsetsToReset = prepareOffsetsToReset(groupId);
+            boolean dryRun = opts.options.has(opts.dryRunOpt) || !opts.options.has(opts.executeOpt);
+            if (!dryRun) {
+                adminClient.alterShareGroupOffsets(groupId,
+                    offsetsToReset.entrySet().stream()
+                        .collect(Collectors.toMap(
+                            Entry::getKey,
+                            entry -> entry.getValue().offset()
+                        )),
+                    new AlterShareGroupOffsetsOptions()
+                        .timeoutMs(opts.options.valueOf(opts.timeoutMsOpt).intValue())
+                ).all().get();
+            }
+            Set<SharePartitionOffsetInformation> partitionOffsets = mapOffsetsToSharePartitionInformation(groupId, offsetsToReset);
+            Map<String, ShareGroupDescription> shareGroups = describeShareGroups(Collections.singleton(groupId));
+            TreeMap<String, Entry<ShareGroupDescription, Collection<SharePartitionOffsetInformation>>> groupOffsets = new TreeMap<>();
+            shareGroups.forEach((id, description) -> groupOffsets.put(id, new SimpleImmutableEntry<>(description, partitionOffsets)));
+            printOffsets(groupOffsets, opts.options.has(opts.verboseOpt));
+        }
+
+        private Map<TopicPartition, OffsetAndMetadata> prepareOffsetsToReset(String groupId) throws ExecutionException, InterruptedException {
```

Comment:
The error handling for `bin/kafka-share-groups.sh --bootstrap-server localhost:9092 --group G1 --reset-offsets --dry-run` needs to be stronger. I didn't specify any topic information so the map `offsetsByTopicPartitions` is empty and the command blows up.